subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/UploadGistsPlugin.java

```diff
@@ -0,0 +1,63 @@
+package org.shipkit.internal.gradle.release;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.shipkit.gradle.configuration.ShipkitConfiguration;
+import org.shipkit.internal.gradle.configuration.ShipkitConfigurationPlugin;
+import org.shipkit.internal.gradle.release.tasks.UploadGistsTask;
+import org.shipkit.internal.gradle.util.StringUtil;
+import org.shipkit.internal.gradle.util.TaskMaker;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Adds a task for uploading files to Gist based on Ant pattern.
+ * <p>
+ *
+ * Applies:
+ * <ul>
+ *     <li>{@link ShipkitConfigurationPlugin}</li>
+ * </ul>
+ *
+ * Adds tasks:
+ * <ul>
+ *     <li>{@link UploadGistsTask}</li>
+ * </ul>
+ */
+public class UploadGistsPlugin implements Plugin<Project> {
+
+    public static final String FILES_PATTERNS_PROPERTY = "filesPatterns";
+
+    private List<String> filesPatterns;
+
+    @Override
+    public void apply(final Project project) {
+        filesPatterns = extractLogDirPatternsFromProjProperty(project);
```

Comment:
Why do we need to expose the project property? Can you describe how this plugin is intended to be used? I'm asking because project property makes the public API bigger.